plugin/trino-oracle/src/main/java/io/trino/plugin/oracle/OracleClient.java

```diff
@@ -705,13 +804,18 @@ public WriteMapping toWriteMapping(ConnectorSession session, Type type)
             }
             return WriteMapping.objectMapping(dataType, longDecimalWriteFunction(decimalType));
         }
-        if (type.equals(TIMESTAMP_SECONDS)) {
-            // Specify 'date' instead of 'timestamp(0)' to propagate the type in case of CTAS from date columns
-            // Oracle date stores year, month, day, hour, minute, seconds, but not second fraction
-            return WriteMapping.longMapping("date", trinoTimestampToOracleDateWriteFunction());
-        }
-        if (type.equals(TIMESTAMP_MILLIS)) {
-            return WriteMapping.longMapping("timestamp(3)", trinoTimestampToOracleTimestampWriteFunction());
+        if (type instanceof TimestampType timestampType) {
+            if (type.equals(TIMESTAMP_SECONDS)) {
+                // Specify 'date' instead of 'timestamp(0)' to propagate the type in case of CTAS from date columns
+                // Oracle date stores year, month, day, hour, minute, seconds, but not second fraction
+                return WriteMapping.longMapping("date", trinoTimestampToOracleDateWriteFunction());
+            }
+            int precision = min(timestampType.getPrecision(), MAX_ORACLE_TIMESTAMP_PRECISION);
+            String dataType = format("timestamp(%d)", precision);
+            if (timestampType.isShort()) {
+                return WriteMapping.longMapping(dataType, oracleTimestampWriteFunction(timestampType));
+            }
+            return WriteMapping.objectMapping(dataType, oracleLongTimestampWriteFunction(createTimestampType(precision)));
```

Comment:
Calling `createTimestampType` looks redundant to me. `oracleLongTimestampWriteFunction` uses only `precision`. Same for `oracleTimestampWriteFunction`

Reply:
want to be consistent with `oracleLongTimestampReadFunction` which requires timestampType